### PR TITLE
Improves spatial receptive field API

### PR DESCRIPTION
**Pull request by Jegp**
Improves spatial receptive fields by providing

* Gradient flows through the parameters
* Sampled and parameterized spatial RFs that allow optimization of various slices of the RF parameters

### CODE REVIEW

File: norse/torch/module/receptive_field.py [hunk 1]
```diff
@@ -4,85 +4,260 @@
...
+                self.scales, self.angles, self.ratios, self.derivatives
+            )
+            self.submodule.has_updated = True
+            self.submodule._update_weights()
+
+
+class ParameterizedSpatialReceptiveField2d(torch.nn.Module):
+    """
+    A parameterized version of the `SpatialReceptiveField2d` module, where the scales, angles, and ratios
+    are optimized and updated during training, if requested.
```
Comment (dimitriskor) [inline norse/torch/module/receptive_field.py:178]: Maybe it should be specified what the difference is with **SampledSpatialReceptiveField2d**
For example:
 ...where the scales, angles, and ratios are optimized and updated for each kernel individually...

File: norse/torch/module/receptive_field.py [hunk 2]
```diff
@@ -4,85 +4,260 @@
...
-        return self.conv(x)
+        self._update_weights()  # Update weights if necessary
+        return torch.nn.functional.conv2d(x, self.weights, bias=None)
+
+
+class SampledSpatialReceptiveField2d(torch.nn.Module):
+    """
+    Creates a spatial receptive field as 2-dimensional convolutions, sampled over a set of scales,
+    angles, ratios, and derivatives.
+    This module allows for the optimization of the scales, angles, and ratios (not derivatives)
```
Comment (dimitriskor) [inline norse/torch/module/receptive_field.py:121]: Maybe it should be specified that the optimization is happening at the input parameters and not at individual kernels
For example:
     This module allows for the optimization of the input parameters of scales, angles, and ratios (not derivatives)

File: norse/torch/module/receptive_field.py [hunk 3]
```diff
@@ -4,85 +4,260 @@
...
-        derivatives (Union[int, List[Tuple[int, int]]]): The number of derivatives to use in the receptive field.
-        aggregate (bool): If True, sums the input channels over all output channels. If False, every
-        output channel is mapped to every input channel, which may blow up in complexity.
-        **kwargs: Arguments passed on to the underlying torch.nn.Conv2d
+    Arguments:
+        in_channels (int): Number of input channels
+        size (int): Size of the receptive field
+        rf_parameters (torch.Tensor): Parameters for the receptive fields
+        aggregate (bool): If `True`, the receptive fields will be aggregated across channels. Defaults to `True`.
+        domain (float): The domain of the receptive field. Defaults to `8`.
```
Comment (dimitriskor) [inline norse/torch/module/receptive_field.py:41]: aggregate and domain parameters are not accessible from SampledSpatialReceptiveField2d and ParameterizedSpatialReceptiveField2d

**Comment by Jegp** [comment]
Excellent comments, thank you @dimitriskor!
